cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractReconciler.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.model.Labels;
+import io.strimzi.operator.common.operator.resource.PvcOperator;
+import io.strimzi.operator.common.operator.resource.StorageClassOperator;
+import io.vertx.core.Future;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Class used for method relevant for reconciliation of Strimzi objects.
+ */
+public abstract class AbstractReconciler {
```

Comment:
I don't think this is the right way. Strimzi has many `Reconciler` classes - in general, we use the term for any class grouping steps for reconciliation of some part of Strimzi. But only two delete PVCs and _extend_ this `AbstractReconciler`. So this makes no sense. 

Moreover - in both of the cases this is used you still have a method for deleting PVC and just call this htourgh `super...` which seems strange.

If you want to remove the duplicate code, I would propose:
* To do it in a separate PR as it does not seem really related to this change
* Instead of creating an abstract class, move the method to some utils class - e.g. `ReconcilerUtils` and call it from the KAfka and ZooKeeper reconcilers.

Reply:
Will revert and move to separate PR.